Removing the main PRB class

wjrosa opened the pull request:
<!--
Did I add a title? A descriptive, yet concise, title.
-->

<!--
Issue: Link to the GitHub issue this PR addresses (if appropriate).
-->

## Changes proposed in this Pull Request:

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for the PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

In this clean-up PR, I am removing the main Payment Request Button backend class (`WC_Stripe_Payment_Request`). It was previously deprecated in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4805.

## Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Please follow the following guidelines when writing testing instructions:

- Include screenshots if there is no similar flow in the critical flows: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Critical-flows
- Assume instructions will be copied over to the Release Testing Instructions wiki page: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions
- Assume instructions will be followed by external testers.
- Assume tester does not have intimate knowledge of Stripe.
-->

Code review should be enough.

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Tested on mobile (or does not apply)

### Changelog entry

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

wjrosa committed:
Removing the main PRB class

wjrosa committed:
Revert unnecessary change

wjrosa committed:
Fix tests

wjrosa committed:
Removing test file

wjrosa committed:
PHPStan baseline file update

wjrosa committed:
Adding the deprecated function

wjrosa committed:
Adding the deprecated function

daledupreez commented on includes/class-wc-stripe-blocks-support.php:
```diff
@@ -194,9 +194,6 @@ public function get_payment_method_data() {
 				'showSaveOption'                  => $this->get_show_save_option(),
 				'isAdmin'                         => is_admin(),
 				'shouldShowExpressCheckoutButton' => $this->should_show_express_checkout_button(),
-				'button'                          => [
-					'customLabel' => $this->payment_request_configuration->get_button_label(),
-				],
```
Should we leave the data structure in place to prevent Javascript runtime errors?
```suggestion
				'button' => [
					'customLabel' => '',
				],
```

AFAICT, the custom label field has been `''` for some time, so it's extremely unlikely that anyone is using it, so I am thinking we probably **don't** want to keep it around.

wjrosa replied:
Good call. Manually applied in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4904/commits/cddfd0887d166948a46973975bffbff2b5976b8b (Github blocked me from applying your suggestion directly)

daledupreez committed:
Merge branch 'develop' into update/removing-the-main-prb-class

daledupreez commented on changelog.txt:
```diff
@@ -1,6 +1,7 @@
 *** Changelog ***
 
 = 10.3.0 - xxxx-xx-xx =
+* Update - Removes the main Payment Request Buttons backend class (previously deprecated)
```
I think we should use `WC_Stripe_Payment_Request` in the description to make it easier to work out what changed.
```suggestion
* Update - Remove the main Payment Request Buttons backend class, WC_Stripe_Payment_Request, which was deprecated in 10.2.0
```

daledupreez commented on includes/class-wc-stripe-blocks-support.php:
```diff
@@ -37,13 +29,21 @@ final class WC_Stripe_Blocks_Support extends AbstractPaymentMethodType {
-	 * @param WC_Stripe_Payment_Request  The Stripe Payment Request configuration used for Payment
-	 *                                   Request buttons.
+	 * @param null                               $payment_request_configuration The Stripe Payment Request configuration used for Payment Request buttons (removed).
+	 * @param WC_Stripe_Express_Checkout_Element $express_checkout_configuration The Stripe Express Checkout configuration used for Express Checkout buttons.
```
The changes to the PHPDoc fixed some existing PHPStan errors, so we need to regenerate the baseline to get those stale error entries removed.

wjrosa replied:
Updated it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4904/commits/327421be4960dd0308fa750dbc0de885332041c0

daledupreez reviewed (changes requested):
The e2e tests are all failing, and I believe it's because there is still an early exit condition in `woocommerce_gateway_stripe_woocommerce_block_support()` where it's checking for `! class_exists( 'WC_Stripe_Payment_Request' )`, and this is preventing Stripe from being available in that context.

daledupreez commented:
I am also wondering whether it may be worth deferring this removal for 10.4.0 in February, as the deprecation only occurred in 10.3.0, and many developers and/or agencies may not have been taking on work between early December and early January, in addition to merchants likely wanting stability over December.

wjrosa committed:
Update changelog.txt

Co-authored-by: daledupreez <dale.du.preez@automattic.com>

wjrosa committed:
Changelog and readme entries update

wjrosa committed:
Baseline file update

wjrosa committed:
Reverting custom label structure removal

wjrosa committed:
Merge branch 'develop' into update/removing-the-main-prb-class

wjrosa committed:
Removing class existence check

wjrosa commented:
> The e2e tests are all failing, and I believe it's because there is still an early exit condition in woocommerce_gateway_stripe_woocommerce_block_support() where it's checking for ! class_exists( 'WC_Stripe_Payment_Request' ), and this is preventing Stripe from being available in that context.

Thanks for the heads up, Dale! I removed the check in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4904/commits/a7ffcaeb5aa5d96a2f3df8cd5a99ba89fe653407

> I am also wondering whether it may be worth deferring this removal for 10.4.0 in February, as the deprecation only occurred in 10.3.0, and many developers and/or agencies may not have been taking on work between early December and early January, in addition to merchants likely wanting stability over December.

Makes sense, and I agree. I have updated the changelog entries and assigned the 10.4.0 milestone here.

daledupreez reviewed (approved):
Thanks for getting those changes done so quickly! 🚀

malithsen reviewed (approved):
Looks good, thanks for the cleanup!

wjrosa committed:
Merge branch 'develop' into update/removing-the-main-prb-class

wjrosa committed:
Updating PHPStan baseline file

wjrosa committed:
Updating PHPStan baseline file

wjrosa committed:
Updating PHPStan baseline file

wjrosa committed:
Updating PHPStan baseline file

wjrosa committed:
Updating version in deprecation warnings
